### ISSUE TITLE
py2exe with new setup / freeze not working with pandas

### ISSUE THREAD

**Issue by schnitzlein** [py2exe/py2exe#167]
Hi,

I invested some time into the topic with py2exe, it is a nice and handy tool.
But unfortunatly for the pandas 1.5.1 or 1.5.2 it show me a ModuleImportError:

> Traceback (most recent call last):
>   File "myfile.py", line 36, in <module>
>   File "pandas\__init__.pyc", line 48, in <module>
>   File "pandas\core\api.pyc", line 47, in <module>
>   File "pandas\core\groupby\__init__.pyc", line 1, in <module>
>   File "pandas\core\groupby\generic.pyc", line 76, in <module>
>   File "pandas\core\frame.pyc", line 171, in <module>
>   File "pandas\core\generic.pyc", line 169, in <module>
>   File "pandas\core\window\__init__.pyc", line 1, in <module>
>   File "pandas\core\window\ewm.pyc", line 15, in <module>
>   File "<loader>", line 15, in <module>
>   File "<loader>", line 13, in __load
> ImportError: (DLL load failed while importing aggregations: ...) 'C:\\Users\\IEUser\\project\\...\\dist\\pandas._libs.window.aggregations.pyd'

I am using this setup:

> from py2exe import freeze
> 
> import numpy, pandas 
> import pylink, usb.core
> 
> sys.setrecursionlimit(3000)
> 
> freeze(
>     console=['project\myfile.py'],
>     windows=[],
>     zipfile='library.zip',
>     options = {
>         "packages": ["matplotlib", "pytz"],
>         "includes": ["numpy", "pandas"]
>         },
>     data_files=None,
>     version_info={}
> )
> 

I check on stackoverflow this issues (looks similar): https://stackoverflow.com/questions/60763529/unable-to-import-pandas-pandas-libs-window-aggregations

but I can't get it running.

I was also first using the outdated/deprecated approach with setuptools, but it leads to the same result.

Python3 test_code file:

myfile.py:
`import pandas as pd

if __name__ == '__main__':
    x = pd.DataFrame()
    print("hello")
`

Also checked the http://www.py2exe.org/index.cgi/FrontPage

**Comment by albertosottile** [comment]
`pandas` is part of our CI, so we check at every commit that `pandas` can be correctly frozen. The last check occurred with version 1.5.1 and it was pass, so I am pretty confident the code of `py2exe` can correctly freeze that version of `pandas`.

Your error seems to point to a missing support DLL in your environment (e.g. MSVC redistributable or .NET framework), so I would advise to check your environment (I would also encourage you to try to run the same code from sources). 

If you want to manually check exactly what DLL is missing, you can run https://github.com/lucasg/Dependencies on the affected file `C:\Users\IEUser\project\...\dist\pandas._libs.window.aggregations.pyd`.

**Comment by albertosottile** [comment]
> I am checking currently the mentioned DLL Tool to find out what Dependencies are missing.
It is a big list, I hope with copy missing *.dll files to dist folder than I am able to run the application.

The dependency walker highlights in red any missing link, the libraries shown in white are already found, there is no need to copy them. 

Perhaps it would be easier if you post here some screenshots from Dependencies.

**Comment by schnitzlein** [comment]
Thanks for your great help.
I started the dependency walker first in command line and it confuses me.
With the GUI tool, I was able to see, all is white and seems to be founded.

Then I install MSVC Redistributable again (https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170) + reboot the System.
Strange but true, I was able to get it running, seems like a simple reboot solved the issue.
